common/src/main/java/org/opensearch/ml/common/output/model/ModelTensors.java

```diff
@@ -41,6 +45,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             }
             builder.endArray();
         }
+        if (statusCode != null) {
```

Comment:
status code are great. But, we need some more details on how client should respond to the status code. Like what are the retryable status code, which are permanent failures?

Another thing, this status code is exact status code from remote model, which is inconvenient from client like neural search. Either we should have ML commons also provide a status code around what it thinks of the error. Because different models can give different status code and then neural search cannot understand each of the model as it doesn't know what it really means and what response should be sent to customer.